### ISSUE TITLE
[Change feed processor] [Deployment units] Resource unit usage 

### ISSUE THREAD

**Issue by shibin-george** [Azure/azure-cosmos-dotnet-v3#3970]
As per [this](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/change-feed-processor?tabs=dotnet#change-feed-and-provisioned-throughput), change feed read operation on the monitored container consume request units.
Say I have a deployment unit, consisting of 5 compute instances. Since, by definition of a deployment unit, all 5 instances are sharing a single lease (assuming one partition), only 1 of the 5 compute instance will be notified of the changes to the monitored container at any given time.
But what about the RU usage in this case? 
- Do all 5 compute instances cause RU usage on the `monitored container` even though only one compute instance is notified at any point of time?
  - In other words, say a change-feed operation costs 5 RUs. Will it cost 5x5=25 RUs when there are are 5 compute-instances sharing the same lease?
  - Also, say a change feed operation costs 5 RUs. If there are 6 deployment-units (i.e. 6 leases) with varying number of compute instances in each dep-unit, would it cost 5*6 = 30 RUs per change feed operation?
- Do all 5 compute instances cause RU usage on the `lease container`?
- Also, is there a max-limit on the number of deployment units that can monitor a container for changes?
  - Any limits on the number of compute-instances with a deployment unit?

Thanks!

**Comment by ealsur** [comment]
> all 5 instances are sharing a single lease (assuming one partition), only 1 of the 5 compute instance will be notified of the changes to the monitored container at any given time.

This is incorrect, they do not share the same lease.

https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/change-feed-processor?tabs=dotnet#dynamic-scaling

> [...] then the change feed processor distributes all the leases that are in the lease container across all running instances of that deployment unit, and it parallelizes compute by using an equal-distribution algorithm. A lease is owned by one instance at any time, so the number of instances shouldn't be greater than the number of leases.

If you have 1 lease and 5 instances, 1 instance will be owning the lease, the rest will be idle.
